src/EncompassRest.Tests/ResourceLocksTest.cs

```diff
@@ -0,0 +1,29 @@
+﻿using EncompassRest.Loans;
+using EncompassRest.ResourceLocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Threading.Tasks;
+
+namespace EncompassRest.Tests
+{
+    [TestClass]
+    public class ResourceLocksTest : TestBaseClass
+    {
+        [TestMethod]
+        public async Task Lock_LoanLockAndUnlock()
+        {
+            var client = await GetTestClientAsync();
+            var loan = new Loan();
+            var loanId = await client.Loans.CreateLoanAsync(loan,true);
+            var locks = await loan.LoanApis.GetLocksAsync();
+            Assert.IsTrue(locks.Count == 0);
+
+            var lockId = await loan.LoanApis.LockAsync(ResourceLockType.Exclusive);
+            var loanLock = await loan.LoanApis.GetLockAsync(lockId);
+            Assert.AreEqual(loanLock.LockType.ToString(), ResourceLockType.Exclusive.ToString());
+
+            Assert.IsTrue(await loan.LoanApis.UnlockAsync(lockId));
```

Comment:
Add test after this to make sure there are no more locks.

Reply:
Still need to retrieve locks to ensure you get one and that it's id matches

Reply:
i do get the lock and verify it.

Reply:
I mean also call `GetLocksAsync`.

Reply:
You might also want to ensure there's only one lock.